src/Microsoft.Health.Fhir.Core/Features/Conformance/SystemConformanceProvider.cs

```diff
@@ -87,6 +87,18 @@ public sealed class SystemConformanceProvider
             _urlResolver = urlResolver;
             _contextAccessor = contextAccessor;
             _searchParameterStatusManager = searchParameterStatusManager;
+
+            // Following code block will log same information multiple times due to Mediatr bug.
+            // Link for Mediatr bug https://github.com/jbogard/MediatR.Extensions.Microsoft.DependencyInjection/issues/34
```

Comment:
Nice find, and you are right, but we have a pattern to fix this.

The registration currently is:
https://github.com/microsoft/fhir-server/blob/662c907ba6ae8625a8555b177fff285c3594a4e9/src/Microsoft.Health.Fhir.Shared.Api/Modules/FhirModule.cs#L131-L134

But can be corrected by following this pattern which removes the Transient handler and re-adds it as a Singleton.

https://github.com/microsoft/fhir-server/blob/662c907ba6ae8625a8555b177fff285c3594a4e9/src/Microsoft.Health.Fhir.Shared.Api/Modules/SearchModule.cs#L86-L91

Reply:
Thanks @brendankowitz for your suggestion. I did above changes in FhirModule.cs and its working. SystemConformanceProvider's constructor doesn't get called twice.

Reply:
Thanks for fixing this. The comment isn't needed now.